api/src/main/java/com/netflix/iceberg/expressions/Literal.java

```diff
@@ -16,87 +16,23 @@
 
 package com.netflix.iceberg.expressions;
 
-import com.netflix.iceberg.types.Type;
 import java.io.Serializable;
-import java.math.BigDecimal;
-import java.nio.ByteBuffer;
 import java.util.Comparator;
-import java.util.UUID;
 
 /**
  * Represents a literal fixed value in an expression predicate
  * @param <T> The Java type of the value wrapped by a {@link Literal}
  */
 public interface Literal<T> extends Serializable {
-  static Literal<Boolean> of(boolean value) {
```

Comment:
I don't have an opinion about whether these type-specific methods should be here or in `Literals`, but I don't see a good reason to remove them in favor of `from` that uses a series of `instanceof` checks.